ouroboros-consensus/src/Ouroboros/Consensus/Storage/LedgerDB/OnDisk.hs

```diff
@@ -17,6 +17,122 @@
 {-# LANGUAGE TupleSections              #-}
 {-# LANGUAGE TypeApplications           #-}
 
+{-|
+  = Snapshot managing and contents
+
+  Snapshotting a ledger state means saving a copy of the in-memory part of the
+  ledger state serialized as a file on disk, as well as flushing differences
+  between the last snapshotted ledger state and the one that we are snapshotting
+  now. While the coupled implementation is in place, the serialized file might
+  also contain a serialized version of the in-memory UTxO map carried by the
+  legacy ledger states, as explained below.
+
+  == Startup
+
+  On startup, the node will:
+
+  1. Find the latest snapshot which will be a directory identified by the slot
+     number of the snapshot:
+
+        > <cardano-node data dir>
+        > ├── volatile
+        > ├── immutable
+        > └── ledger
+        >     ├── <slotNumber1>
+        >     │   ├── tables
+        >     │   └── state
+        >     ├── <slotNumber2>
+        >     │   ├── tables
+        >     │   └── state
+        >     └── <slotNumber3>
+        >         ├── tables
+        >         └── state
+
+  2. Depending on the snapshots found, there are two possibilities:
+
+       - If there is no snapshot to load, create a new @'BackingStore'@ with the
+         contents of the Genesis UTxO and finish.
+
+       - If there is a snapshot found, then deserialize the @state@ file, which
+         contains either:
+
+            - If the node was running __with the legacy database enabled__, it
+              contains a full LedgerState snapshot in the same style as before
+              the UTxO HD implementation (i.e. in-memory part of the LedgerState
+              (equal among implementations) + UTxO map).
+
+            - Otherwise, it contains a serialization of the in-memory part of the
+              LedgerState, without a UTxO map.
+
+        In both cases, the snapshot will be deserialized as @ExtLedgerState blk
+        EmptyMK@. In case we found an snapshot, we will overwrite the
```

Comment:
`we will overwrite`: eg using `rm -f tables; cp <snapshot>/tables -r .` is one option that ought to work. But I think instead the backing store implementation should be the one to create copies and load in/reset to/restore from a copy. EG some backing store (eg RocksDB, if I recall correctly) might be able to only store a "restore point" identifier in the snapshot's `tables` file and the restore operation would just be a more efficient reset to the identified former state.

"we will overwrite" might be simple enough wording, but only if it isn't taken literally.